File: src/main/java/com/ericsson/oss/ci/domain/dto/configuration/Job.java
/*******************************************************************************
 * COPYRIGHT Ericsson 2021
 *
 *
 *
 * The copyright to the computer program(s) herein is the property of
 *
 * Ericsson Inc. The programs may be used and/or copied only with written
 *
 * permission from Ericsson Inc. or in accordance with the terms and
 *
 * conditions stipulated in the agreement/contract under which the
 *
 * program(s) have been supplied.
 ******************************************************************************/
package com.ericsson.oss.ci.domain.dto.configuration;

import com.ericsson.oss.ci.domain.enumeration.JobTypeEnum;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Job {
    private String jenkinsUrl;
    private String jobName;
    private String appType;
    private Set<String> products;
    private JobTypeEnum type;
    
    public Job(String job){
    	this.jobName = job;
    }
}

File: src/main/web/application/src/models/job.ts
///
/// COPYRIGHT Ericsson 2021
///
///
///
/// The copyright to the computer program(s) herein is the property of
///
/// Ericsson Inc. The programs may be used and/or copied only with written
///
/// permission from Ericsson Inc. or in accordance with the terms and
///
/// conditions stipulated in the agreement/contract under which the
///
/// program(s) have been supplied.
///


import { ConditionalStageStatus } from './cell/cell-stage';
import { Stage } from './stage';

export class Job {
  [x: string]: any;
  jobName!: string;
  jobUrl!: string;
  buildNo!: string;
  buildUrl!: string;
  stageAlignedOrNot!: boolean;
  helmVersion?: string
  cbosVersion?: string;
  currentCbosVersion!: boolean;
  mandatoryStages?: Stage[];
  unknownStages?: Stage[];
  skippedStages?: Stage[];
  lastSuccessBuildDuration!: string;
  lastSuccessBuildTimestamp!: string;
  buildStagePercentage!: string;
  jobproductList?: string[];
  sonarMetrics?:Metric[];
  sonarQubeUrl?: string;
  sonarQualityGateStatus?: string;
  sonarReportStatusCode?: number;
  conditionalStages: ConditionalStageResponse = {};
}

export type ConditionalStageResponse =  {[key: string] : ConditionalStageStatus };

export class Metric{
  metric!: string;
  value!: string;
  period!: any;
}